pkg/client/cli/intercept_state.go

```diff
@@ -275,30 +301,61 @@ func (is *interceptState) createRequest() (*connector.CreateInterceptRequest, er
 
 	spec.Agent = is.agentName
 	spec.TargetHost = "127.0.0.1"
+
 	// Parse port into spec based on how it's formatted
 	portMapping := strings.Split(is.port, ":")
-	switch len(portMapping) {
-	case 1:
-		port, err := strconv.ParseUint(is.port, 10, 16)
+	portError := func() error {
+		if is.dockerRun {
+			return errors.New("ports must be of the format --ports <local-port>:<container-port>[:<svcPortIdentifier>]")
+		}
+		return errors.New("ports must be of the format --ports <local-port>[:<svcPortIdentifier>]")
+	}
+
+	parsePort := func(portStr string) (uint16, error) {
+		port, err := strconv.ParseUint(portStr, 10, 16)
 		if err != nil {
-			return nil, errors.Errorf("Port numbers must be a valid, positive int, you gave: %q", is.port)
+			return 0, fmt.Errorf("port numbers must be a valid, positive int, you gave: %q", is.port)
 		}
-		spec.TargetPort = int32(port)
+		return uint16(port), nil
+	}
+
+	port, err := parsePort(portMapping[0])
+	if err != nil {
+		return nil, err
+	}
+	is.localPort = port
+	spec.TargetPort = int32(port)
+
+	switch len(portMapping) {
+	case 1:
 	case 2:
-		port, err := strconv.ParseUint(portMapping[0], 10, 16)
-		if err != nil {
-			return nil, errors.Errorf("Port numbers must be a valid, positive int, you gave: %q", portMapping[0])
+		if port, err = parsePort(portMapping[1]); err == nil && is.dockerRun {
```

Comment:
This might just be the case of my first time seeing this, but will we hit the else if `err != nil`? If so, I think we may want to split this out to error check correctly if err != nil && is.dockerRun is false. Or is there some special behavior when checking err inline that the else only looks at the 2nd condition